spec/test_spec.lua

```diff
@@ -76,3 +78,173 @@ describe("luarocks test #integration", function()
       end)
    end)
 end)
+
+test_env.unload_luarocks()
+
+local fs = require("luarocks.fs")
+local path = require("luarocks.path")
+local test = require("luarocks.test")
+local test_busted = require("luarocks.test.busted")
+local test_command = require("luarocks.test.command")
+
+describe("LuaRocks test #unit", function()
+   local runner
+   
+   setup(function()
+      runner = require("luacov.runner")
+      runner.init(testing_paths.testrun_dir .. "/luacov.config")
+      runner.tick = true
+   end)
+   
+   teardown(function()
+      runner.shutdown()
+   end)
+   
+   local tmpdir
+   local olddir
+   
+   local create_tmp_dir = function()
+      tmpdir = get_tmp_path()
+      olddir = lfs.currentdir()
+      lfs.mkdir(tmpdir)
+      lfs.chdir(tmpdir)
+      fs.change_dir(tmpdir)
+   end
+   
+   local destroy_tmp_dir = function()
+      if olddir then
+         lfs.chdir(olddir)
+         if tmpdir then
+            lfs.rmdir(tmpdir)
+         end
+      end
+   end
+   
+   describe("test.command", function()
+      describe("command.detect_type", function()
+         before_each(function()
+            create_tmp_dir()
+            write_file("test.lua", "", finally)
+         end)
+         
+         after_each(function()
+            destroy_tmp_dir()
+         end)
+         
+         it("returns true if test.lua exists", function()
+            assert.truthy(test_command.detect_type())
```

Comment:
write the file inside this test instead of `before_each`. this allows the tests to run more independently of each other (and in any order).